js/languagePickerModel.js

```diff
@@ -46,6 +56,79 @@ define([
             }
 
             this.set('_languages', languages);
+        },
+            
+        onDataLoaded: function() {
+            _.defer(function() {
+                this.locationId = Adapt.offlineStorage.get('location') || null;
+                this.restoreState();
+            }.bind(this));
+
+        },
+        
+        restoreLocation: function() {
+            _.defer(function() {
+                if (this.locationId !== "undefined" && this.locationId) {
```

Comment:
move this check up and return early? 

also `"undefined"` is a slightly unusual value to check for, did you find circumstances under which it gets set to this?